es6-lib/decoders/layer.js

```diff
@@ -267,6 +268,12 @@ class Layer extends Duplex {
 
         if(!self.push(sep + rowString + ep)) {
           this.pause();
+
+          if (!self._readableState.pipes) {
```

Comment:
might as well copy & paste the explanatory comment here too, unless there's a clever way of sharing this logic between all three (?)